cmd/kubeadm/app/util/version.go

```diff
@@ -204,3 +210,31 @@ func kubeadmVersion(info string) (string, error) {
 	vStr := fmt.Sprintf("v%d.%d.%d%s", v.Major(), v.Minor(), patch, pre)
 	return vStr, nil
 }
+
+// Validate if the remote version is one Minor release newer than the client version.
+// This is done to conform with "stable-X" and only allow remote versions from
+// the same Patch level release.
+func validateStableVersion(remoteVersion, clientVersion string) (string, error) {
+	if clientVersion == "" {
+		glog.Infof("could not obtain client version; using remote version: %s", remoteVersion)
+		return remoteVersion, nil
+	}
+
+	verRemote, err := versionutil.ParseGeneric(remoteVersion)
```

Comment:
In theory, here we have content of remote file. right now, it is supposed to be exact version. but the idea of recursive validation was that if label 'stable-1' will be resolved to another label, e.g. 'stable-1.12', this function will try to resolve one more remote label. If this function will expect that remote content is always parsable version, it will break previous functionality.

Reply:
thanks for the review but this was in flight for 10 days.
also just created the cherry pick for 1.12.

i see what you are saying but in my tests i wasn't able to reach such a breakage.
we do fall-back to `stable-1.12`, if `stable-1` is much newer, recurse and it works fine.

now, if `stable-1` returns a non-semantic version as contents, that's really a breakage with the endpoint from from my perspective.

Reply:
Sorry for delay with commenting on that, somehow screwed up notifications on this PR :(

Yes, that scenario is not covered by tests. It was previously implicitly working with nesting of the labels. Support for nested label resolution is not critical, but nice to have. I can do separate PR for that later on.

Reply:
sounds good! thanks.